hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconPipelineReportHandler.java

```diff
@@ -63,8 +63,16 @@ protected void processPipelineReport(PipelineReport report,
     PipelineID pipelineID = PipelineID.getFromProtobuf(report.getPipelineID());
     if (!reconPipelineManager.containsPipeline(pipelineID)) {
       LOG.info("Unknown pipeline {}. Trying to get from SCM.", pipelineID);
-      Pipeline pipelineFromScm =
-          scmServiceProvider.getPipeline(report.getPipelineID());
+      Pipeline pipelineFromScm = null;
+
+      try {
+        pipelineFromScm =
+            scmServiceProvider.getPipeline(report.getPipelineID());
+      } catch (PipelineNotFoundException pnfe) {
```

Comment:
From SCMClient, I don't think we get PipelineNotFoundException I believe we get exception as IOException.

```

  public static IOException getRemoteException(ServiceException se) {
    Throwable e = se.getCause();
    if (e == null) {
      return new IOException(se);
    }
    return e instanceof IOException ? (IOException) e : new IOException(se);
  }
```
```

  private ScmContainerLocationResponse submitRequest(
      StorageContainerLocationProtocolProtos.Type type,
      Consumer<Builder> builderConsumer) throws IOException {
    final ScmContainerLocationResponse response;
    try {
      Builder builder = ScmContainerLocationRequest.newBuilder()
          .setCmdType(type)
          .setVersion(CURRENT_VERSION)
          .setTraceID(TracingUtil.exportCurrentSpan());
      builderConsumer.accept(builder);
      ScmContainerLocationRequest wrapper = builder.build();

      response = submitRpcRequest(wrapper);
    } catch (ServiceException ex) {
      throw ProtobufHelper.getRemoteException(ex);
    }
    return response;
  }
```

Can you add a simple test with a cluster start, and see do we get PipelineNotFoundException?

Reply:
please see the log i mention [above ](https://github.com/apache/ozone/pull/2371#discussion_r658695134), the server side throw a PipelineNotFoundException, so if i do not misunderstand, the client side will also get a PipelineNotFoundException. please correct me if i am wrong

Reply:
````
    try {
      cluster.getStorageContainerLocationClient().getPipeline(HddsProtos.PipelineID.newBuilder().setId(UUID.randomUUID().toString()).build());
    }  catch (IOException ex) {
      Assert.assertTrue(ex instanceof RemoteException);
      PipelineNotFoundException exception = (PipelineNotFoundException) ((RemoteException) ex).unwrapRemoteException(PipelineNotFoundException.class);
      Assert.assertTrue(exception instanceof PipelineNotFoundException);
    }
````

We donot get PipelineNotFoundException, we get RemoteException wrapped with PipelineNotFoundException. We need to unwrap and check if it is PipelineNotFoundException.

Reply:
thanks for the review ,  i will fix this